lib/input_data_page.dart

```diff
@@ -0,0 +1,141 @@
+                        bloc: _bloc),
+                    const SizedBox(height: 15),
+                    DatePick(
+                        widgetName: Constants.toDateNameString,
+                        pick: state.lastDate,
+                        bloc: _bloc),
+                  ]);
+                } else {
+                  return const Text('Unknown error');
+                }
+              }),
+          const SizedBox(height: 15),
+          ElevatedButton(
+            onPressed: () {
+              if (_bloc.dateOfBirth != DateTime(0000)) {
+                _bloc.calculateAge();
+                Navigator.push(
+                    context,
+                    MaterialPageRoute(
+                        builder: (_) => BlocProvider.value(
+                              value: _bloc,
+                              child: CalculatedDataPage(),
+                            )));
+              } else {
+                ScaffoldMessenger.of(context).showSnackBar(
+                  SnackBar(
+                    content: const Text('Fill in the From date field!'),
+                  ),
+                );
+              }
+            },
+            child: const Text('Calculate Age'),
+          )
+        ],
+      ),
+    );
+  }
+}
+
+class DatePick extends StatelessWidget {
+  const DatePick(
+      {required this.widgetName, required this.pick, required this.bloc});
+
+  final String widgetName;
+  final DateTime pick;
+  final InputDataBloc bloc;
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Text(widgetName, style: const TextStyle(fontSize: 17)),
+        Container(
+          decoration: BoxDecoration(
+            border: Border.all(
+              color: Colors.black,
+              width: 3,
+            ),
+            borderRadius: BorderRadius.circular(12),
+          ),
+          child: Padding(
+            padding: const EdgeInsets.all(8.0),
+            child: Row(
+              children: [
+                const SizedBox(
+                  width: 25,
+                ),
+                InkWell(
+                    onTap: () {
+                      _showPicker(context, widgetName).then((date) {
+                        if (widgetName == Constants.fromDateNameString) {
+                          bloc.addFirstDateOfBirthEvent(date!);
+                        } else if (widgetName == Constants.toDateNameString) {
+                          bloc.addSecondDateOfBirthEvent(date!);
+                        }
+                      });
+                    },
+                    child: Text(pick == DateTime(Constants.defaultDate)
+                        ? 'Pick up a date'
+                        : pick.toString())),
+              ],
+            ),
+          ),
+        ),
+      ],
+    );
+  }
+
+  Future<DateTime?> _showPicker(BuildContext context, String widgetName) {
```

Comment:
extract to above mentioned separate file

Reply:
extracted to separate date_pick_widget.dart file